cmd/observer/node_builder/observer_builder.go

```diff
@@ -1206,6 +1243,25 @@ func (builder *ObserverServiceBuilder) BuildExecutionSyncComponents() *ObserverS
 
 			return nil
 		}).
+		Module("events storage", func(node *cmd.NodeConfig) error {
+			builder.events = store.NewEvents(node.Metrics.Cache, node.ProtocolDB)
+			return nil
+		}).
+		Module("execution state cache", func(node *cmd.NodeConfig) error {
+			// TODO: use real objects instead of mocks once they're implemented
+			snapshot := osyncsnapshot.NewSnapshotMock(
+				builder.events,
+				nil,
+				nil,
```

Comment:
the observer doesn't have transactions and collections? It should be able to index them using the indexer